### PR TITLE
Adds amp-toolbox-cache-list

### PR DESCRIPTION
**Pull request by andreban**
Retrieves the list of know AMP Caches from `https://cdn.ampproject.org/caches.json`

### CODE REVIEW

File: cache-list/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Caches = require('./lib/AmpCaches.js');
+module.exports = Caches;
```
Comment (sebastianbenz) [inline cache-list/index.js]: Can you document the exports (see http://usejsdoc.org/tags-module.html)

File: cache-list/lib/AmpCaches.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {OneBehindFetch} = require('amp-toolbox-core');
+const CACHE_LIST_ENDPOINT = 'https://cdn.ampproject.org/caches.json';
+
+class AmpCaches {
```
Comment (sebastianbenz) [inline cache-list/lib/AmpCaches.js]: Please add JSDoc

File: cache-list/lib/AmpCaches.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  constructor(fetchStrategy = OneBehindFetch.create()) {
+    this.fetchStrategy_ = fetchStrategy;
+  }
+
+  list() {
+    return this.getCaches_();
+  }
+
+  async get(cacheName) {
```
Comment (sebastianbenz) [inline cache-list/lib/AmpCaches.js]: What's cacheName? the id?

File: cache-list/spec/CachesSpec.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          expect(cacheList.length).toBe(1);
+        });
+    });
+  });
+
+  describe('get', () => {
+    it('returns the correct cache', () => {
+      caches.get('google')
+        .then(googleCache => {
+          expect(googleCache.id).toBe('google');
```
Comment (sebastianbenz) [inline cache-list/spec/CachesSpec.js]: A single expect should be sufficient:

```
expect(googleCache).toBe(CACHES_JSON.caches[0]);
```

File: cache-list/lib/AmpCaches.js (hunk 2 above)
Comment (andreban) [reply cache-list/lib/AmpCaches.js]: done.

File: cache-list/lib/AmpCaches.js (hunk 3 above)
Comment (andreban) [reply cache-list/lib/AmpCaches.js]: renamed to cacheId.

File: cache-list/spec/CachesSpec.js (hunk 4 above)
Comment (andreban) [reply cache-list/spec/CachesSpec.js]: done.

File: cache-list/index.js (hunk 1 above)
Comment (andreban) [reply cache-list/index.js]: Added a doc. Not sure if that's what you meant.

**Comment by andreban** [comment]
PTAL.

File: cache-list/lib/AmpCaches.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * List of known AMP Caches, as available at `https://cdn.ampproject.org/caches.json`.
+ *
+ */
+class AmpCaches {
+  /**
+   * Creates a new instance of AmpCaches.
+   *
+   * @param {Object} fetchStrategy The fetch strategy to be used when fetching
```
Comment (sebastianbenz) [inline cache-list/lib/AmpCaches.js]: is there a way to indicate this as optional?

Comment (andreban) [reply cache-list/lib/AmpCaches.js]: per JSDoc: 

```javascript
@param {number} [foo=1]`
```
 so: 
```javascript
@param {Object} [fetchStrategy = OneBehindStrategy]
```

**Pull request merged**